libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -701,8 +704,11 @@ async def _astream(
             else:
                 yield cast(ChatGenerationChunk, result.generations[0])
             return
-        async with self._async_client.messages.stream(**params) as stream:
-            async for text in stream.text_stream:
+        stream = await self._async_client.messages.create(**params, stream=True)
+        async for event in stream:
+            # See https://github.com/anthropics/anthropic-sdk-python/blob/main/src/anthropic/lib/streaming/_messages.py  # noqa: E501
+            if event.type == "content_block_delta" and event.delta.type == "text_delta":
```

Comment:
I think at the very least the `message_delta` event type might contain usage information:

https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-anthropic/src/chat_models.ts#L701

Reply:
yes, will follow in a separate PR

Reply:
added to this PR actually.